joblib/test/test_parallel.py

```diff
@@ -648,6 +645,50 @@ def test_direct_parameterized_backend_context_manager():
     assert _active_backend_type() == DefaultBackend
 
 
+@with_multiprocessing
+def test_nested_backend_context_manager():
+    # Check that by default, nested parallel calls will always use the
+    # ThreadingBackend
+
+    def get_nested_pids():
+        assert _active_backend_type() == ThreadingBackend
+        return Parallel(n_jobs=2)(delayed(os.getpid)() for _ in range(2))
+
+    for backend in ['threading', 'loky', 'multiprocessing']:
+        with parallel_backend(backend):
+            pid_groups = Parallel(n_jobs=2)(
+                delayed(get_nested_pids, check_pickle=False)()
+                for _ in range(10)
+            )
+            for pid_group in pid_groups:
+                assert len(set(pid_group)) == 1
+
+
+@with_multiprocessing
+def test_retrieval_context():
+    import contextlib
+
+    class MyBackend(ThreadingBackend):
+        i = 0
+
+        @contextlib.contextmanager
+        def retrieval_context(self):
+            self.i += 1
+            yield
+
+    register_parallel_backend("retrieval", MyBackend)
+
+    def nested_call(n):
+        return Parallel(n_jobs=2)(delayed(id)(i) for i in range(n))
+
+    with parallel_backend("retrieval") as (ba, _):
+        Parallel(n_jobs=2)(
+            delayed(nested_call, check_pickle=False)(i)
```

Comment:
We can probably remove that `check_pickle=False` now that loky is the default backend and that cloudpickle is used to pickle function definitions.